### PR TITLE
Combined work sprint1

### PR DESCRIPTION
**Pull request by clecardona**
#### Backend : 
- all following endpoints implemented (see below)
image

#### Frontend : 
-Work of all week, design

### CODE REVIEW

**Review by sangeetha-murugesan-sda9 — APPROVED** [review]

**Review by DakouriKobri — APPROVED** [review]

**Review by sangeetha-murugesan-sda9 — APPROVED** [review]

**Review by sangeetha-murugesan-sda9 — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -163,39 +229,38 @@ Network Trash Folder
...
+# Windows Installer files
+*.cab
+*.msi
+*.msix
+*.msm
+*.msp
+
+# Windows shortcuts
+*.lnk
+
```
Comment (sangeetha-murugesan-sda9) [inline .gitignore:265]: Reviewed

**Review by sangeetha-murugesan-sda9 — APPROVED** [review]

File: src/main/java/backend/user/ResourceNotFoundException.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package backend.user;
+import org.springframework.http.HttpStatus;
```
Comment (sangeetha-murugesan-sda9) [inline src/main/java/backend/user/ResourceNotFoundException.java:2]: Reviewed

**Review by sangeetha-murugesan-sda9 — APPROVED** [review]

File: .gitignore [hunk 3]
```diff
@@ -1,7 +1,76 @@
...
+.scala_dependencies
+.worksheet
+
+# Uncomment this line if you wish to ignore the project description file.
+# Typically, this file would be tracked if it contains build/dependency configurations:
+#.project
+
+### Eclipse Patch ###
+# Spring Boot Tooling
+.sts4-cache/
```
Comment (sangeetha-murugesan-sda9) [inline .gitignore:73]: Reviewed

**Review by sangeetha-murugesan-sda9 — APPROVED** [review]

**Review by sangeetha-murugesan-sda9 — APPROVED** [review]

**Review by sangeetha-murugesan-sda9 — APPROVED** [review]
ok for me

**Review by sangeetha-murugesan-sda9 — APPROVED** [review]

**Review by sangeetha-murugesan-sda9 — APPROVED** [review]

File: .gitignore [hunk 4]
```diff
@@ -1,7 +1,76 @@
-<<<<<<< HEAD
-=======
-# Created by https://www.toptal.com/developers/gitignore/api/intellij,gradle,react,java,macos
-# Edit at https://www.toptal.com/developers/gitignore?templates=intellij,gradle,react,java,macos
+
+# Created by https://www.toptal.com/developers/gitignore/api/intellij,eclipse,java,windows,macos,vscode
```
Comment (sangeetha-murugesan-sda9) [inline .gitignore:2]: ok

**Review by sangeetha-murugesan-sda9 — APPROVED** [review]

**Review by sangeetha-murugesan-sda9 — APPROVED** [review]

File: .gitignore [hunk 5]
```diff
@@ -1,7 +1,76 @@
...
+.target
+
+# Tern plugin
+.tern-project
+
+# TeXlipse plugin
+.texlipse
+
+# STS (Spring Tool Suite)
+.springBeans
```
Comment (sangeetha-murugesan-sda9) [inline .gitignore:53]: Reviewed.

File: .gitignore [hunk 6]
```diff
@@ -163,39 +229,38 @@ Network Trash Folder
...
+# Recycle Bin used on file shares
+$RECYCLE.BIN/
+
+# Windows Installer files
+*.cab
+*.msi
+*.msix
+*.msm
+*.msp
+
```
Comment (sangeetha-murugesan-sda9) [inline .gitignore:262]: Reviewed

File: .vscode/settings.json [hunk 7]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (sangeetha-murugesan-sda9) [inline .vscode/settings.json:1]: Reviewed

File: README.md [hunk 8]
```diff
@@ -1,2 +1,5 @@
 # SDA-group-project
```
Comment (sangeetha-murugesan-sda9) [inline README.md:1]: Reviewed

File: build.gradle [hunk 9]
```diff
@@ -26,9 +26,16 @@ dependencies {
...
     developmentOnly 'org.springframework.boot:spring-boot-devtools'
     compile group: 'com.auth0', name: 'java-jwt', version: '3.8.3'
+    implementation group: 'commons-io', name:'commons-io', version:'2.6'
     runtimeOnly 'org.postgresql:postgresql'
 }
 
+
 test {
     useJUnitPlatform()
 }
```
Comment (sangeetha-murugesan-sda9) [inline build.gradle:36]: Reviewed

File: frontend/src/assets/img/logo/instyle.svg [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+   xmlns:inkscape="http://www.inkscape.org/namespaces/inkscape"
+   width="150mm"
+   height="150mm"
+   viewBox="0 0 150 150"
+   version="1.1"
+   id="svg8"
+   inkscape:version="1.0.2 (e86c8708, 2021-01-15)"
+   sodipodi:docname="instyle.svg">
+  <defs
+     id="defs2" />
```
Comment (sangeetha-murugesan-sda9) [inline frontend/src/assets/img/logo/instyle.svg:18]: Reviewed

File: frontend/src/assets/img/logo/instyle.svg [hunk 11]
```diff
@@ -0,0 +1,98 @@
...
+     inkscape:window-width="1440"
+     inkscape:window-height="847"
+     inkscape:window-x="0"
+     inkscape:window-y="25"
+     inkscape:window-maximized="1"
+     units="mm" />
+  <metadata
+     id="metadata5">
+    <rdf:RDF>
+      <cc:Work
```
Comment (sangeetha-murugesan-sda9) [inline frontend/src/assets/img/logo/instyle.svg:42]: Reviewed

File: frontend/src/components/Navbar.jsx [hunk 12]
```diff
@@ -36,6 +37,10 @@ export default function Navbar({ onLogout }) {
           <button className="btn-menu" ><img src= {homeButton} alt="menu"/></button>
         </Popover>
       </div>
+
```
Comment (sangeetha-murugesan-sda9) [inline frontend/src/components/Navbar.jsx:40]: Reviewed

File: frontend/src/pages/WinnerPage/WinnerPage.jsx [hunk 13]
```diff
@@ -3,7 +3,7 @@ import React from "react";
 import "../../styles/base.css"
 import NavBar from "../../components/Navbar"
 import UserMeta from "../../components/UserMeta"
```
Comment (sangeetha-murugesan-sda9) [inline frontend/src/pages/WinnerPage/WinnerPage.jsx:5]: Reviewed

File: frontend/src/styles/_buttons.css [hunk 14]
```diff
@@ -59,6 +59,8 @@
 }
 
 .btn-menu{
+  height: 50px;
```
Comment (sangeetha-murugesan-sda9) [inline frontend/src/styles/_buttons.css:62]: Reviewed

File: frontend/src/styles/_layout.css [hunk 15]
```diff
@@ -24,6 +24,8 @@ p{
 
 h1{
     text-align: center;
+    font-size: var(--font-M);
```
Comment (sangeetha-murugesan-sda9) [inline frontend/src/styles/_layout.css:27]: Reviewed

File: frontend/src/styles/_nav.css [hunk 16]
```diff
@@ -1,13 +1,15 @@
 
 
 nav{
+    
     width:80%;
     display: grid ;
-    grid-template-columns: 60px 1fr ;
+    grid-template-columns: 60px 60px 60px 1fr ;
```
Comment (sangeetha-murugesan-sda9) [inline frontend/src/styles/_nav.css:7]: Reviewed

File: frontend/src/styles/_nav.css [hunk 17]
```diff
@@ -47,4 +49,12 @@ nav{
        font-weight: 600;
        font-size:var(--font-L) ;
```
Comment (sangeetha-murugesan-sda9) [inline frontend/src/styles/_nav.css:50]: Reviewed

File: frontend/src/styles/_structure.css [hunk 18]
```diff
@@ -16,13 +16,14 @@ header{
 
   display: flex;
```
Comment (sangeetha-murugesan-sda9) [inline frontend/src/styles/_structure.css:17]: Reviewed

File: frontend/src/styles/_vote_component.css [hunk 19]
```diff
@@ -1,4 +1,5 @@
 .vote-container{
```
Comment (sangeetha-murugesan-sda9) [inline frontend/src/styles/_vote_component.css:1]: Reviewed

File: frontend/src/styles/_winner.css [hunk 20]
```diff
@@ -1,19 +1,22 @@
 .winner-content{
+        
         z-index: 1;      
         height: 100%;     
 
         display: grid;
-        grid-auto-rows: 30px min-content min-content 100px ;
```
Comment (sangeetha-murugesan-sda9) [inline frontend/src/styles/_winner.css:6]: Reviewed

File: gradle/wrapper/gradle-wrapper.properties [hunk 21]
```diff
@@ -3,3 +3,6 @@ distributionPath=wrapper/dists
 distributionUrl=https\://services.gradle.org/distributions/gradle-6.7-bin.zip
 zipStoreBase=GRADLE_USER_HOME
```
Comment (sangeetha-murugesan-sda9) [inline gradle/wrapper/gradle-wrapper.properties:4]: Reviewed

File: src/main/java/backend/SkeletonApplication.java [hunk 22]
```diff
@@ -1,7 +1,11 @@
 package backend;
 
 import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
```
Comment (sangeetha-murugesan-sda9) [inline src/main/java/backend/SkeletonApplication.java:4]: Reviewed

File: src/main/java/backend/auth/CustomUserDetailsService.java [hunk 23]
```diff
@@ -3,6 +3,7 @@
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.security.core.userdetails.UserDetails;
 import org.springframework.security.core.userdetails.UsernameNotFoundException;
```
Comment (sangeetha-murugesan-sda9) [inline src/main/java/backend/auth/CustomUserDetailsService.java:5]: Reviewed

**Review by sangeetha-murugesan-sda9 — APPROVED** [review]

**Comment by sangeetha-murugesan-sda9** [comment]
Reviewed

**Review by sangeetha-murugesan-sda9 — APPROVED** [review]

**Review by Ayatj — APPROVED** [review]
Good job

**Pull request merged**